### ISSUE TITLE
Vendr Custom Rewards Not Being Applied

### ISSUE THREAD

**Issue by philipdanielhayton** [vendrhub/vendr#4]
Hi guys,

Not sure if this is the right place to ask so I've also posted it on the Umbraco Discord...

I have a client that's on Umbraco 11.4.2 / Vendr v3.0.15. Unfortunately upgrading to Umbraco Commerce is not an option at this point.

We are trying to setup a custom discount reward that deducts the order subtotal from the shipping costs. For example, if the order subtotal is 40 EUR and the shipping costs 60 EUR, then the discount rule should deduct 40 EUR from the shipping cost.

I have created a custom DiscountRewardProvider to handle this but it's not actually applying to the cart. During checkout, the totals are still showing their original amount.

There are a couple of things that make this a little bit more perplexing.
(1) I can see that the adjustments are being created in the database table [vendrOrderPriceAdjustment], but they're not being reflected on the final order
(2) During testing I tried implementing the adjustment using a custom PriceAdjuster and it initially worked as intended, but if I mix and match the Reward and the PriceAdjuster methods - both of them stop working

I'm not sure whether I'm missing an important step somewhere, or if this is a bug. Does anybody have any experience with this?

For info, here's an example of my RewardProvider code, using some hard coded silly values for testing purposes:

```
public override DiscountRewardCalculation CalculateReward(DiscountRewardContext context, MinDeliveryAmountDiscountRewardProviderSettings settings)
{
   var result = new DiscountRewardCalculation();
   var price = new Price(999, 0, context.Order.CurrencyId);
   result.ShippingTotalPriceAdjustments.Add(new DiscountAdjustment(context.Discount, adjustedPrice));
   return result;
}
```

Any help is greatly appreciated :pray_tone1:

**Comment by mattbrailsford** [comment]
Initial issue I can see is your discount price is a positive amount when it should be a negative, but this said the code snippet doesn't look like something that will work anyway as `adjustedPrice` isn't defined anywhere.

Can you try updating the example to something that can be used to replicate this?

**Comment by philipdanielhayton** [comment]
Hi @mattbrailsford 

Sorry your right, the example shown above didn't work, I changed my code to make it more readable in the example outside of my code editor and didn't re-test.

Anyway, I did more digging and found the culprit. It was in my own code, but it's something that could probably do with highlighting in the docs - just to help other developers avoid the same pitfall.

Tldr; I was pulling the order subtotal from context.Order.SubtotalPrice, when I should have been using context.OrderCalculation.SubtotalPrice. This was throwing my logic off and _somtimes_ causing it to return too early. 🤦🏻‍♂️

**Comment by mattbrailsford** [comment]
No problem, glad you found it 👍🏻

And just to confirm there are some docs around this here https://docs.umbraco.com/umbraco-commerce/key-concepts/order-calculation-state but whether we should link that from another location in the docs I’m not quite sure 🤔
